File: jscript/f30.loop.js
// spin loop as condition is always true
//  use break to exit out of loop
do {
  // prompt user
  WScript.stdout.write("Enter your name (quit to Exit): ");
  answer = WScript.stdin.readline();         // get input
  if (answer == "quit")
      break                                  // exit loop is user quits
  else	  
      WScript.echo("Hello " + answer + "!"); // output result as not exiting
} while (true)

File: ruby/f30.loop.rb
#!/usr/bin/env ruby
# pure spin loop using continous loop construct
#   break is used to exit loop
loop do
  print "Enter you name (quit to Exit): " # print prompt
  STDOUT.flush                            # flush buffer to show prompt
  answer = gets.chomp                     # get string input
 
  if answer.chomp == "quit"               # check for exit
      break                               # exit loop
  end

  puts "Hello #{answer}!"                 # print result as not exiting
end